explorer/e2e/testFunctions.ts

```diff
@@ -515,11 +535,101 @@ export async function testClearAll(
   for (let i = 0; i < filterNames.length; i++) {
     await page.getByText(filterRegex(filterNames[i])).dispatchEvent("click");
     await expect(
-      getNamedFilterButtonLocator(page, selectedFilterNamesList[i]).getByRole(
+      getNamedFilterOptionLocator(page, selectedFilterNamesList[i]).getByRole(
         "checkbox"
       )
     ).not.toBeChecked();
     await page.locator("body").click();
   }
 }
+
+/**
+ * Escape a string so it can safely be used in a regexp
+ * @param string - the string to escape
+ * @returns - A string that has all Regexp special characters escaped
+ */
+function escapeRegExp(string: string): string {
+  return string.replace(/[.*+?^${}()|[\]\\]/g, "\\$&");
```

Comment:
Nice! Would you be able to add a comment describing the "why" behind the pattern used here?

Reply:
Will do! (it's copy-pasted from Findable lol)